knimin/lib/tests/test_data_access.py

```diff
@@ -297,6 +297,99 @@ def test_get_ag_barcode_details(self):
             del(obs[key]['registered_on'])
         self.assertEqual(obs, exp)
 
+    def test_plate_operations(self):
+        """Test four functions involving read/write operations to tables
+        pm.sample and pm.plate
+
+        Notes
+        -----
+        They are put together in one test, because they are dependent on each
+        other.
+        """
+        # setUp
+        sample_ids = ['test_sample_1', 'test_sample_2', 'test_sample_3']
+        db._add_new_samples(sample_ids)
+
+        # Test set_plate_info
+        # Create new plate
+        plate_info = {
+            'name': 'test plate 01',
+            'email': 'test',
+            'plate_type_id': 1,  # 96-well
+            'linker_seq': 'GT',
+            'extraction_robot_id': 1  # HOWE_KF1
+        }
+        obs = db.set_plate_info(0, plate_info)
+        self.assertGreater(obs, 0)
+
+        # Edit existing plate
+        plate_id = obs
+        plate_info = {
+            'processing_robot_id': 2,  # RIKE
+            'tm300_8_tool_id': 3,  # 109375A
+        }
+        obs = db.set_plate_info(plate_id, plate_info)
+        exp = plate_id
+        self.assertEqual(obs, exp)
+
+        # Test get_plate_info:
+        exp = {
```

Comment:
I understand your rationale. However, I am thinking that the front end will have to do additional calls to retrieve actual values anyway, because the drop-down menus needs items. Therefore returning ids and letting the front end find and display actual values seems to be a better solution. Maybe there are other ways to do this?